main.go

```diff
@@ -363,20 +374,23 @@ func getFromMetadata(urlStr string) ([]byte, error) {
 		},
 	}
 	resp, err := client.Do(req)
+	defer resp.Body.Close()
```

Comment:
Why is this change required?

According to the `http` package documentation for `Do()`:
```
On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when
CheckRedirect fails, and even then the returned Response.Body is already closed.
```